### ISSUE TITLE
Add support for multiple inputs per mode in EMAP

### ISSUE THREAD

**Issue by gchhablani** [pliang279/MultiViz#14]
The current implementation assumes that the `predictor_fn` will take only one input per mode and then the combinations are calculated based on those two inputs.

However, there can be cases where a single input mode has multiple inputs.

An example for LXMERT:

```python

test_question = [test_question]

    inputs = lxmert_tokenizer(
        test_question,
        padding="max_length",
        max_length=20,
        truncation=True,
        return_token_type_ids=True,
        return_attention_mask=True,
        add_special_tokens=True,
        return_tensors="pt"
    )

    output_vqa = lxmert_vqa(
        input_ids=inputs.input_ids,
        attention_mask=inputs.attention_mask,
        visual_feats=features,
        visual_pos=normalized_boxes,
        token_type_ids=inputs.token_type_ids,
        output_attentions=False,
    )
```

Here, LXMERT needs two inputs for visual mode: `normalized_boxes` and `features`. For text, it needs three inputs: `inputs.attention_mask`, `inputs.input_ids` and `inputs.token_type_ids`.

Current implementation will take all these inputs and for combinations for all. So, for example, a particular `normalized_boxes` won't correspond to the correct instance of `features`. This is problematic.

In order to fix this, we can add either `tuple` or `dict` inputs in the dataset dict.

An example dataset with dict inputs will look like:

```python
dataset = {
    'visual_inputs': {
        'features': <numpy array>,
        'normalized_boxes': <numpy array>
    },
    'textual_inputs':{
        'input_ids': <numpy array>,
        'attention_mask': <numpy array>,
        'token_type_ids': <numpy array>,

    }
}     
```
and with tuples:

```python
dataset = {
    'visual_inputs': (
        <numpy array for features>,
        <numpy array for normalized_boxes>
    ),
    'textual_inputs':(
        <numpy array for input_ids>,
        <numpy array for attention_mask>,
        <numpy array for token_type_ids>
    )
}
```

Providing support for both would be great. And, a similar pattern can be followed for the upcoming analysis methods.
